# Analysis of City of Seattle Staff Demographic Data
Ziah Daily

09/23/19

## Abstract

In the file is dataset which includes all current City of Seattle Employees, their age, their race/ethnicity, their sex, the dapartment they work in, their hourly rate, their status as an employee, and whether or not they are a regular or temporary worker. Having this data is important becuase it allows us to refelct on the people that work for the city. Additonally, it rasies questions about minimum, maximum, and average age and wage of each city employee. 
For this lab I am taking the list of all of their age and comapring it to all of their wages.


## Dataset Preparation


I got this file from https://data.seattle.gov/City-Business/City-of-Seattle-Staff-Demographics/5avq-r9hj. This dataset has a total of 14.1K rows and 7 collumns. According to the city of Seattle, this dataset was created February 25, 2019, but last updated September 26,2019.

I am writing the code below becuase I want to analyze the data on the City of Seattle's Staff Demographics. The first line of code allows me to to open the code on jupeter notebooks. The second line creates an empty list to store data from the file that we opened on the City of Seattles Staff Demographics. The third line of code is putting all lines from the file into a list. The fourth line of Data is closing the demographic file, closing the file is a corteous and clean thing to do. The last line is a representation of what the first 5 rows of code look like, giving people an idea of what it looks like.

In [46]:
#list of all the names
#list of all wages

#open the file
demographic_file = open("City_of_Seattle_Staff_Demographics.csv", "r");

#create an empty list to store the data from the csv file
demographic_data = []; 

#put all lines from the file into a list
for row in demographic_file:
    demographic_data.append(row);
    
#close the file. It's the courteous and clean thing to do
demographic_file.close();

#this is what five rows of the data look like-helps give people an idea of what it looks like
print(demographic_data[:5]);


['Race/Ethnicity,Sex,Department,Age,Hourly Rate,Regular/Temporary,Employee Status\n', 'Hispanic or Latino,M,Municipal Court,57,29.75,R,A\n', 'Hispanic or Latino,F,Municipal Court,39,27.18,R,A\n', 'Hispanic or Latino,M,Police Department,52,59.75,R,A\n', 'Hispanic or Latino,M,Fire Department,30,38.28,R,A\n']


^^^The list printed above represnts the first five rows in the file demographic_data. I printied the first five rows in order to let people know what the data looks like

## Data Modeling

In this data modeling setion I am extracting two differnt columns and making them into their own seperate lists.

In [47]:
#max, min, average, median ,and desriptions

import statistics

#reopen file
#demographic_file = open("City_of_Seattle_Staff_Demographics.csv", "r");

#remove the first element (column header)
# the header is useful for telling us what the columns mean, 
# but you don't want to include it in your mathematical calculations
del demographic_data[0];

# create two empty lists - one for age and wage
# we want to make a list of just the age and the wage 
age = [];
wage = [];  

# loop (go through every row one by one) through the original demographic_data
for item in range(len(demographic_data)):
    
    # this line is a bit confusing, but what it is doing is splitting out each comma separated item 
    # from a row and turning it into a list 
    # for example, this line of code turns he single string '9/1/15,1393,1483\n' 
    # into three separate strings '9/1/15' '1393' and '1483'
    # After being split, the three separate strings are then stored in another list called demographic_row
    demographic_row = demographic_data[item].split(",")
    
    #now that we've split the row into its three parts, we can pick and choose 
    # add the second and the third to specific lists for west and east
    age.append(int(demographic_row[3]))
    wage.append(float(demographic_row[4]))


Now that I have my two differnt list, I am going to make sime simple calculations on the minium, maximum, average, and median for both of my lists.

In [48]:
    
agemin = min(age)

agemax = max(age)

ageavg = statistics.mean(age)

agemed = statistics.median(age)

wagemin = min(wage)

wagemax = max(wage)

wageavg = statistics.mean(wage)

wagemed = statistics.median(wage)


#print all the data and their titles 
print("age minimum:")
print(agemin) 
print("wage minimum:")
print(wagemin)
print("age maximum:")
print(agemax)
print("wage maximum")
print(wagemax)
print("age average:")
print(round(ageavg))
print("wage average:")
print(wageavg)
print("age medium:")
print(agemed)
print("wage medium:")
print(wagemed)

age minimum:
14
wage minimum:
5.53
age maximum:
92
wage maximum
162.8353
age average:
45
wage average:
40.03381942370015
age medium:
45.0
wage medium:
39.02


## Data Analysis and Conclusion

### conclusions:

In conclusion this data set shows that the minuimum age of an someone that is working for the city of seattle is 14 years old. This is extremly suprising and raises a lot of questions. In additon to having the minimum age I also calculated the average and maximum wage. The maximum age is 92 years old and the avergae age is 45 years old. After calculating the age, I decided that I would calculate the wage. Similar to age, I learned the maximum, minimum, and average of wage. The minimum wage is 5.53 dollars, the maximum wage is 162 dollars per hour, and the average wage is about 40 dollars an hour.

### questions raised:

Why is the yougest age 14?

Is having a 14 year old working for thecity llegal?

Why is there such  a large age gap?

Why is there such a large wage gap? 



## Acknowledgments

For the first Lab on the analyis of the City of Seattle Demographic Data, I got assistence from Ms. Sconyers and Navya.